### PR TITLE
Add confirmation popup when dragging in image

### PR DESCRIPTION
**Pull request by Karthik99999**
https://www.smogon.com/forums/threads/background-image-dragging-dropping.3695386/

Demo of how it works

https://user-images.githubusercontent.com/32044378/148441191-39a391f5-35da-4ca9-806e-c782e94b74bc.mp4

### CODE REVIEW

File: js/client-topbar.js [hunk 1]
```diff
@@ -805,23 +805,40 @@
...
 		};
 		reader.readAsDataURL(file);
 	};
 
+	var ConfirmBackgroundPopup = this.ConfirmBackgroundPopup = Popup.extend({
+		initialize: function (data) {
+			var buf = '';
+			buf += '<p>Are you sure you want to change your background to this image?</p>';
+			buf += '<p><img src="' + data.bgUrl + '" style="display:block;margin:auto;max-width:90%"></p>';
+			buf += '<p class="buttonbar"><button name="setBg" value="' + data.bgUrl + '"><strong>Confirm</strong></button> ';
```
Comment (Zarel) [inline js/client-topbar.js:830]:
```suggestion
			buf += '<p class="buttonbar"><button name="setBg" value="' + data.bgUrl + '"><strong>Change background</strong></button> ';
```

Comment (Zarel) [reply js/client-topbar.js:830]: (Design principle D2)

File: js/client-topbar.js [hunk 2]
```diff
@@ -805,23 +805,40 @@
...
-			Storage.bg.set(e.target.result, 'custom', noSave);
 		};
 		reader.readAsDataURL(file);
 	};
 
+	var ConfirmBackgroundPopup = this.ConfirmBackgroundPopup = Popup.extend({
+		initialize: function (data) {
+			var buf = '';
+			buf += '<p>Are you sure you want to change your background to this image?</p>';
+			buf += '<p><img src="' + data.bgUrl + '" style="display:block;margin:auto;max-width:90%"></p>';
```
Comment (Zarel) [inline js/client-topbar.js:829]: Possibly also a max-height if you accidentally drag a super tall but thin image, so the "Cancel" button isn't way off in Narnia.

**Comment by Zarel** [comment]
Thanks! This is exactly what I wanted!

File: js/client-topbar.js (hunk 1 above)
Comment (Karthik99999) [reply js/client-topbar.js:830]: Did you want me to remove the question at the top as well?

Comment (Karthik99999) [reply js/client-topbar.js:830]: @Zarel?

Comment (Zarel) [reply js/client-topbar.js:830]: Up to you. It's not a big deal either way.

Comment (Zarel) [reply js/client-topbar.js:830]: I'd lean towards removing the question, though.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**